pkg/core/template/engine.go

```diff
@@ -50,18 +50,30 @@ func (e *Engine) Register(spec Spec) {
 	e.TemplateSpecs[spec.Type] = spec
 }
 
-func (e *Engine) RenderTemplate(templateType config.TemplateItemType, context map[string]interface{}, option RenderOptions) (out string, err error) {
-	templateBody, spec, err := e.resolveTemplate(templateType, option)
+func (e *Engine) RenderTemplate(templateType config.TemplateItemType, context map[string]interface{}, resolveOptions ResolveOptions, validateOpts ...func(*Validator)) (out string, err error) {
+	templateBody, spec, err := e.resolveTemplate(templateType, resolveOptions)
 	if err != nil {
 		return
 	}
 	if spec.IsHTML {
-		return RenderHTMLTemplate(string(templateType), templateBody, context)
+		return RenderHTMLTemplate(RenderOptions{
+			Name:          string(templateType),
+			TemplateBody:  templateBody,
+			Defines:       spec.Defines,
```

Comment:
When would `Defines` be used?

Reply:
To break the elements of Auth UI into several parts. And reuse them in the main page.

Reply:
Can user influence the content of `Defines` list? If yes, a billion laugh attack is possible.

Reply:
> Can user influence the content of Defines list?

No. The Defines list is defined by Spec which is a compile time constant.